server-common/src/main/java/org/apache/kafka/server/common/CheckpointFile.java

```diff
@@ -72,7 +71,7 @@ public CheckpointFile(File file,
         tempPath = Paths.get(absolutePath.toString() + ".tmp");
     }
 
-    public void write(Collection<T> entries) throws IOException {
+    public void write(List<T> entries) throws IOException {
```

Comment:
Why did we change this?

Reply:
This is done to make write and read API consistent like below.

```
public void write(List<T> entries) throws IOException 
public List<T> read() throws IOException 
```

Reply:
I don't think consistency is necessarily a goal in cases like this. The original intent was to accept the minimal interface required, but to return something more specific to avoid unnecessary conversions.

Reply:
Sure, I am fine with the suggestion for now. I updated PR with the changes to keep it like earlier. But this kind of contract looks fragile, we can take a relook at whether we can avoid any conversions and keep the APIs consistent. Filed https://issues.apache.org/jira/browse/KAFKA-14625 to followup later.